app/app.go

```diff
@@ -922,10 +1028,30 @@ func NewEvmos(
 	return app
 }
 
+// CheckTx will check the transaction with the provided checkTxHandler. We override the default
+// handler so that we can verify bid transactions before they are inserted into the mempool.
+// With the POB CheckTx, we can verify the bid transaction and all of the bundled transactions
+// before inserting the bid transaction into the mempool.
+func (app *Evmos) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
+	return app.checkTxHandler(req)
+}
+
+// SetCheckTx sets the checkTxHandler for the app.
+func (app *Evmos) SetCheckTx(handler mev_lane.CheckTx) {
+	app.checkTxHandler = handler
+}
+
+// ChainID gets chainID from private fields of BaseApp
+// Should be removed once SDK 0.50.x will be adopted
+func (app *Evmos) ChainID() string {
+	field := reflect.ValueOf(app.BaseApp).Elem().FieldByName("chainID")
+	return field.String()
+}
```

Comment:
Instead of using `reflect`, we can get the chainID from a flag or the config file. The same as in `[root.go](https://github.com/evmos/evmos/blob/0e8101d5927733db6ecf0fee1fd03d8eb6dfd87a/cmd/evmosd/root.go#L273-L288)`, e.g.:

```
home := cast.ToString(appOpts.Get(flags.FlagHome))
	chainID := cast.ToString(appOpts.Get(flags.FlagChainID))
	if len(chainID) == 0 {
		v := viper.New()
		v.AddConfigPath(filepath.Join(home, "config"))
		v.SetConfigName("client")
		v.SetConfigType("toml")
		if err := v.ReadInConfig(); err != nil {
			panic(err)
		}
		conf := new(clientconfig.ClientConfig)
		if err := v.Unmarshal(conf); err != nil {
			panic(err)
		}
		chainID = conf.ChainID
	}
```

Maybe we can define this helper function and reutilize it